lib/http2/http2_debug_state.c

```diff
@@ -86,19 +86,19 @@ __attribute__((format(printf, 3, 4))) static void append_chunk(h2o_mem_pool_t *p
 
 static void append_header_table_chunks(h2o_mem_pool_t *pool, h2o_iovec_vector_t *chunks, h2o_hpack_header_table_t *header_table)
 {
-    int i, comma_removed = 0;
+    int i;
     for (i = 0; i < header_table->num_entries; i++) {
         h2o_hpack_header_table_entry_t *entry = h2o_hpack_header_table_get(header_table, i);
         append_chunk(pool, chunks,
-                    ",\n"
-                    "      [ \"%.*s\", \"%.*s\" ]",
+                    "\n"
+                    "      [ \"%.*s\", \"%.*s\" ],",
                     (int)entry->name->len, entry->name->base,
                     (int)entry->value->len, entry->value->base);
-        if (!comma_removed) {
-            chunks->entries[chunks->size - 1].base[0] = ' ';
-            comma_removed = 1;
-        }
     }
+
+    if (i > 0)
+        // remove the last commna
+        --chunks->entries[chunks->size - 1].len;
```

Comment:
Please do not insert a comment taking a line within a if statement (without using a block), since it is counterintuitive.

You should write the comment either _above_ the `if` keyword or surround the comment and the statement that is executed if `if` evaluates to true using `{}`.